packages/react/src/floating-ui-react/hooks/useTypeahead.ts

```diff
@@ -194,7 +200,27 @@ export function useTypeahead(context: FloatingRootContext, props: UseTypeaheadPr
     }
   });
 
-  const reference: ElementProps['reference'] = React.useMemo(() => ({ onKeyDown }), [onKeyDown]);
+  const onBlur = useEventCallback((event: React.FocusEvent) => {
+    const next = event.relatedTarget as Node | null;
+    const withinReference = next ? elements.domReference?.contains(next) === true : false;
+    const withinFloating = next ? elements.floating?.contains(next) === true : false;
```

Comment:
These should use shadow DOM `contains` util

Reply:
Okay, I'll check it out as soon as possible